packages/node/src/services.ts

```diff
@@ -172,11 +189,23 @@ class FirebaseStoreService implements IStoreService {
     return result;
   }
 
-  async set(pairs: { key: string; value: any }[]): Promise<any> {
+  /**
+   * Bulk set operation. Note that while firebase specifies that null/undefined
+   * values are interpreted as a delete for the key
+   * (https://www.firebase.com/docs/web/api/firebase/set.html), we throw an
+   * error by defautl instead.
+   */
+  async set(
+    pairs: { key: string; value: any }[],
+    allowDelete?: Boolean
+  ): Promise<any> {
     const updates = {};
     for (const pair of pairs) {
       updates[pair.key] = JSON.parse(JSON.stringify(pair.value));
     }
+    if (!allowDelete && containsNull(updates)) {
+      throw new Error("Firebase store service found null/undefined value");
```

Comment:
This description is a little misleading since the store service itself hasn't found `null` or `defined` - we're just not writing these values to the store by choice.

So maybe, "The records being set contain `null` values" or something like that.

Reply:
And maybe add this as an ERROR constant to the set of errors we have, so we don't copy the string literal as we're doing in the test.

Reply:
Also this directly throws - I thought the suggestion was to warn and not error?

Reply:
I think developers naturally ignore warnings. Since there's a switch anyway, it should error unless explicitly told not to.

Reply:
Makes sense - I think the comment about the error message still stands tho

Reply:
fixed